neuraxle/steps/loop.py

```diff
@@ -157,38 +157,19 @@ def __init__(self, wrapped: BaseStep, copy_op=copy.deepcopy, cache_folder_when_n
         self.steps_as_tuple: List[NamedTupleList] = []
         self.copy_op = copy_op
 
-    def set_train(self, is_train: bool = True):
-        MetaStepMixin.set_train(self, is_train)
-        [step.set_train(is_train) for _, step in self]
-        return self
-
-    def set_hyperparams(self, hyperparams: HyperparameterSamples) -> BaseStep:
-        MetaStepMixin.set_hyperparams(self, hyperparams)
-        self.steps_as_tuple = [(name, step.set_hyperparams(self.wrapped.get_hyperparams())) for name, step in self]
-        return self
-
-    def update_hyperparams(self, hyperparams: HyperparameterSamples) -> BaseStep:
-        """
-        Update the step hyperparameters without removing the already-set hyperparameters.
-        Please refer to :func:`~BaseStep.update_hyperparams`.
+    def apply(self, method_name: str, step_name=None, *kargs, **kwargs) -> Dict:
```

Comment:
Please document the inputs and outputs in docstring and explain what it do. I have difficulty reading the code and understanding the types and logic without more comments. Adding typing as well here would be good. Please consider that I always require typing in core classes and related classes.